datafusion/functions-aggregate/src/min_max.rs

```diff
@@ -914,69 +888,56 @@ impl Accumulator for SlidingMaxAccumulator {
     }
 }
 
-/// MIN aggregate expression
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub struct Min {
-    name: String,
-    data_type: DataType,
-    nullable: bool,
-    expr: Arc<dyn PhysicalExpr>,
+    signature: Signature,
+    aliases: Vec<String>,
 }
 
 impl Min {
-    /// Create a new MIN aggregate function
-    pub fn new(
-        expr: Arc<dyn PhysicalExpr>,
-        name: impl Into<String>,
-        data_type: DataType,
-    ) -> Self {
+    pub fn new() -> Self {
         Self {
-            name: name.into(),
-            expr,
-            data_type: min_max_aggregate_data_type(data_type),
-            nullable: true,
+            signature: Signature::user_defined(Volatility::Immutable),
+            aliases: vec!["min".to_owned()],
         }
     }
 }
 
-impl AggregateExpr for Min {
-    /// Return a reference to Any that can be used for downcasting
-    fn as_any(&self) -> &dyn Any {
+impl Default for Min {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl AggregateUDFImpl for Min {
+    fn as_any(&self) -> &dyn std::any::Any {
         self
     }
 
-    fn field(&self) -> Result<Field> {
-        Ok(Field::new(
-            &self.name,
-            self.data_type.clone(),
-            self.nullable,
-        ))
+    fn name(&self) -> &str {
+        "MIN"
```

Comment:
Follow up PR is to lowercase the name

Reply:
I logged this https://github.com/apache/datafusion/issues/11779 I am not sure we should lowercase this or make the other UDF uppercase